src/__tests__/customFileParserTest/customFileParserTest.ts

```diff
@@ -0,0 +1,191 @@
+    expect(parsedFile.arrays.z_translations.length).toBe(0);
+    checkFloatArrayMatch(parsedFile.arrays.z_translations.slice(0, 20), expectedZTranslations);
+
+    const expectedGeometryIndexes: any = {
+      1: [7060329795587960, 1043, 1, 36, 34, 47, 0, 0, 0, 10, 0, 2920898236136174, 2420, 4, 97, 707, 50, 0, 3, 4],
+      2: [8164540613673614, 2631, 2, 313, 112, 13, 3, 10, 2332139652741056, 3137, 2, 48, 258, 1, 2, 10,
+        5768382459793153, 1355, 1, 106],
+      3: [927342382680797, 2778, 2, 15, 287, 2, 2, 0, 4, 16, 2412602947335905, 3516, 2, 80, 105, 9, 11, 14, 8, 6],
+      4: [5869005083795563, 3346, 2, 3, 121, 0, 1, 7, 34, 7392096689693635, 3167, 2, 75, 260, 1, 1, 150, 2,
+        83056106740777, 4630],
+      13: [1776196732743416, 1625, 1, 61, 853, 0, 2, 40, 65, 0, 2556907504057857, 375, 1, 774, 73, 0, 4, 40, 65, 5],
+      14: [2409675612804526, 2419, 4, 97, 708, 16, 1, 4, 21, 20, 1396465678319174, 1569, 1, 406, 120, 3, 3, 4, 21, 20],
+      15: [8922988493723012, 2092, 5, 55, 158, 90, 0, 54, 24, 1828017180807764, 2172, 4, 4, 63, 0, 2, 9, 1,
+        2317576119257839, 2171],
+      19: [9000406793629980, 2969, 2, 2, 518, 221, 32, 258, 0, 0, 73, 0, 9, 15, 15, 4, 1],
+      21: [5829254376003860, 1433, 1, 20, 453, 181, 3, 0, 3, 0, 1, 6050315401479636, 1435, 1, 20, 411, 34, 4, 0, 3],
+      22: [1080060884278809, 4305, 3, 22, 52, 77, 0, 43, 30, 7160316963937309, 2962, 2, 161, 618, 10, 1, 4, 0,
+        6626953872348268, 4931],
+      100: [4820468089466959, 3425, 2, 2, 2373107104903371, 1340, 40, 1, 2740906351331585, 3084, 2, 2,
+        7453191589183843, 4760, 12, 3, 7022047152622673, 1243, 20, 1],
+    };
+
+    for (let k = 0; k < parsedFile.geometries.length; k++) {
+      const type = parsedFile.geometries[k].type;
+      const indexes = parsedFile.geometries[k].indexes;
+      for (let m = 0; m < Math.min(20, expectedGeometryIndexes[type].length); m++) {
+        expect(indexes.nextValue() === expectedGeometryIndexes[type][m]);
+      }
+    }
+  });
+
+  test('generate groups', async() => {
+    const incomingFile = fs.readFileSync('./src/__tests__/customFileParserTest/Pipes.c3d', null);
+
+    const asArrayBuffer = new ArrayBuffer(incomingFile.length);
+    const arrayBufferCopier = new Uint8Array(asArrayBuffer);
+    for (let i = 0; i < incomingFile.length; i++) {
+      arrayBufferCopier[i] = incomingFile[i];
+    }
+
+    const parsedFile = parseCustomFileFormat(asArrayBuffer, true);
+    const geometryGroups = generateGeometryGroups(parsedFile);
+
+    expect(geometryGroups).toBeDefined();
+
+    expect(geometryGroups.boxGroup.capacity).toBe(327);
+    expect(geometryGroups.circleGroup.capacity).toBe(251 + 2 * 435 + 2 * 913);
+    expect(geometryGroups.coneGroup.capacity).toBe(435 + 913);
+
+    const boxes = geometryGroups.boxGroup;
+    expect(boxes.getNodeId(0)).toBe(7060329795587960);
+    expect(boxes.getTreeIndex(0)).toBe(1043);
+    // checkIntArrayMatch(boxes.getColor(), [1, 36, 34, 47]);
```

Comment:
Should this be commented out?

Reply:
Oh yeah, I don't actually know what that value is supposed to be, so I'll just take it out